spec/Candid.md

```diff
@@ -144,10 +144,12 @@ A *service* is a standalone actor on the platform that can communicate with othe
 
 #### Structure
 
-A service's signature is described by an *actor type*, which defines the list of *methods* that the service provides. Each method is described by its *name* and a *function type* describing its signature. The function type can also be given by referring to a type definition naming a function reference type.
+A service's signature is described by an *actor type*, which defines the list of *methods* that the service provides. 
+A service can optionally provide *initialization parameters* for creating services dynamically with the same signature.
```

Comment:
Hm, it may be good to make the difference more explicit. The parameters aren't just "optional". If they are present, the type describes a different kind of entity. That probably deserves a separate paragraph.

Reply:
I wonder if `{ <methtype>;* }` should just be a sugar for `() -> { <methtype>;* }`? So the presence of parameters represents different stages of the service: uninitialized vs installed

Reply:
In what context? In general, they are totally different things, the former typing a canister reference, the latter a Wasm module. They are no more related than `Nat` and `() -> Nat`.

Reply:
In the sense that the did file generated by Motoko is only a Wasm module. It becomes a canister only after installation. So the actor from Motoko is always `() -> service`.